File: frontend/prest/src/app/models/restaurant/RestaurantUpdatePayload.ts
import {Payload} from "../common/Payload";
import {UserRestaurant} from "@userRestaurant/UserRestaurant";
import {Category} from "@category/Category";
import {Address} from "@address/Address";

export interface RestaurantUpdatePayload extends Payload
{
  id_restaurant: number;
  label: string;
  description: string;
  menu: String;
  categories: Category[];
  actif: boolean;
  address: Address;
  image: String;
  userRestaurants: UserRestaurant[];
}

File: Prest/src/main/java/be/prest/Payload/Update/RestaurantUpdatePayload.java
package be.prest.Payload.Update;

import be.prest.Entities.Address;
import be.prest.Entities.Category;
import be.prest.Entities.UserRestaurant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.web.multipart.*;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RestaurantUpdatePayload {
    private int id_restaurant;
    private String label;
    private String description;
    private MultipartFile menu;
    List<Category> categories;
    private boolean actif;
    private MultipartFile image;
    private List<UserRestaurant> userRestaurants;
    private Address address;

}
